src/OpenTelemetry.Extensions.Docker/Resources/DockerResourceDetector.cs

```diff
@@ -60,9 +62,28 @@ internal Resource BuildResource(string path)
     /// <summary>
     /// Extracts Container Id from path.
     /// </summary>
+    /// /// <param name="path">cgroup path.</param>
+    /// <returns>Returns Resource with list of key-value pairs of container resource attributes if container id exists else empty resource.</returns>
+    private string ExtractContainerId(string path)
+    {
+        try
+        {
+            return this.ExtractContainerIdV1(path) ?? this.ExtractContainerIdV2(path);
```

Comment:
just thinking out loud.
If the cgroup path will match the Id V2 (e.g. "13:name=systemd:/pod/d86d75589bf6cc254f3e2cc29debdf85dde404998aa128997a819ff991827356/hostname") the only way ExtractContainerIdV1() won't return you "hostname" is (I assume, please correct me if I'm wrong) because of EncodingUtils.IsValidHexString() that won't treat this string valid.
So my thoughts:
1. it's a bit too advanced in the V1 logic to finally decide that this is the wrong string (I mean it goes thru a lot of code before realizing this)
2. it's a bit fragile and is very easy to make a bug because from a IsValidHexString() context it's not clear how important this code for this place here. I know it's covered by unit tests but still

So maybe since the V2 logic puts a very specific requirement on the format of the string we should run it first? It will early out much faster than V1

Reply:
I agree with that logic. After reviewing it, I believe that the IsValidHexString() is the method that prevents the hostname from being accepted and switching the two would be helpful since, as you mentioned, it has a stricter requirement. One way to make IsValidHexString() less fragile is to perhaps make a more strict requirement like a character limit (64 characters)? But it is also unclear to me how crucial this method is so I definitely do not think that I should change it here.